Xamarin.Forms.Core/OrientationStateTrigger.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using Xamarin.Forms.Internals;
+using FormsDevice = Xamarin.Forms.Device;
+
+namespace Xamarin.Forms
+{
+	public class OrientationStateTrigger : StateTriggerBase
+	{
+		public OrientationStateTrigger()
+		{
+			UpdateState();
+
+			if (!DesignMode.IsDesignModeEnabled)
+			{
+				var weakEvent = new WeakEventListener<OrientationStateTrigger, object, EventArgs>(this)
+				{
+					OnEventAction = (instance, source, eventArgs) => OnSizeChanged(source, eventArgs),
```

Comment:
DeviceInfo implements INPC so you can just subscribe to that and watch for Orientation changes